cmd/plume/prerelease.go

```diff
@@ -257,17 +257,36 @@ func getCLImageFile(client *http.Client, src *storage.Bucket, fileName string) (
 	return imagePath, nil
 }
 
+
+
 func getFedoraImageFile(client *http.Client, spec *channelSpec, src *storage.Bucket, fileName string) (string, error) {
 	cacheDir := filepath.Join(sdk.RepoCache(), "images", specChannel, specVersion)
 	rawxzPath := filepath.Join(cacheDir, fileName)
 	imagePath := strings.TrimSuffix(rawxzPath, ".xz")
 
+	imageBuildDir := spec.BuildDir
+	imageMetadata := imageMetadataAbstract{
+		Env:       specEnv,
+		Version:   specVersion,
+		Timestamp: specTimestamp,
+		Respin:    specRespin,
+		ImageType: specImageType,
+		Arch:      specBoard,
+	}
+
+	t := template.Must(template.New("builddir").Parse(imageBuildDir))
+	buffer := &bytes.Buffer{}
+	if err := t.Execute(buffer, imageMetadata); err != nil {
+		return "", err
+	}
+	imageBuildDir = buffer.String()
+
 	if _, err := os.Stat(imagePath); err == nil {
 		plog.Printf("Reusing existing image %q", imagePath)
 		return imagePath, nil
 	}
 
-	rawxzURI, err := url.Parse(fmt.Sprintf("%v/%v/compose/%v/%v/images/%v", spec.BaseURL, specComposeID, getImageTypeURI(), specBoard, fileName))
+	rawxzURI, err := url.Parse(fmt.Sprintf("%v/%v/compose/%v/%v/images/%v", spec.BaseURL, imageBuildDir, getImageTypeURI(), specBoard, fileName))
```

Comment:
Only the `user` & `cloud` channels have `spec.BuildDir` not empty. This probably needs to be gated to use the templated `imageBuildDir` on those channels while still using `specComposeID` on others.

Reply:
Per our DM on IRC, I removed the Fedora Atomic Host channels and added `spec.BuildDir ` to the `branched` channel.